### PR TITLE
feat: protect Markdown code regions from typographic transformations

### PR DESCRIPTION
**Pull request by eunice-hong**
## Summary

Extend the tokenizer to recognize Markdown inline code spans and fenced code blocks, and exclude their content from all typographic transformations — no configuration required.

## Background / Motivation

SmartyPants is commonly applied to Markdown content (editors, static site generators, documentation tools). Until now, text inside inline code (`` `a->b` ``) and fenced code blocks (` ``` `) was subject to the same typographic rules as regular prose, silently corrupting literal code.

For example:

```markdown
Use `a->b` to indicate direction.
```

```
x != y
```

would incorrectly produce `` `a→b` `` and `x ≠ y` inside the code regions. Developers relying on smartypants for Markdown rendering had no safe way to prevent this without pre-stripping code blocks themselves.

## Related Issues

- Closes #

## Changes

- **`TokenType.markdown`** — new enum value for code-region tokens; masked with the same U+FFFC placeholder mechanism used for HTML, so their content is never passed through `_applyTransformations`.
- **`tokenize()`** — detects `` ` `` and `~` as additional trigger characters and delegates to `scanMarkdown()`.
- **`_Scanner.scanMarkdown()`** — character-by-character scanner that handles single/double-backtick inline spans, triple-or-more backtick/tilde fenced blocks, and graceful fallback for unclosed spans.
- **`_Scanner._scanFencedBlock()`** — shared helper for backtick and tilde fences; closes on the first line starting with ≥ `openCount` fence characters, or protects to end of input if unclosed.
- **`smartypants_base.dart`** — masking condition extended to include `TokenType.markdown` alongside `TokenType.html`.
- **`example/lib/example_data.dart`** — new "Markdown Support" category with illustrative examples.
- **`benchmark/smartypants_benchmark.dart`** — benchmark cases for light and heavy Markdown inputs.

Markdown protection is always active; no API flag is needed because backticks and tildes are essentially never used outside of code contexts in prose.

## Test Plan

- Commands run:
    - `dart test`
    - `cd example && flutter test`
- Notes:
    - 28 new tests in `test/smartypants_markdown_test.dart` covering inline code, backtick fences, tilde fences, unclosed regions, edge cases, and mixed HTML + Markdown.
    - All 148 tests (existing + new) pass with no regressions.

## Documentation (If Needed)

- Dartdoc on `TokenType`, `tokenize()`, `scanMarkdown()`, and `_scanFencedBlock()` updated to reflect Markdown token support.
- Example app "Markdown Support" category demonstrates the protection in action.

## Breaking Change (If Any)

- `TokenType` gains a new `markdown` value. Code that exhaustively switches on `TokenType` without a default case will need a new branch.
- Migration notes:
    - Add a `markdown` case (or a default) to any exhaustive `switch` on `TokenType`. Treat it the same as `html` — protected content that should not be modified.

### CODE REVIEW

File: lib/src/tokenizer.dart [hunk 1]
```diff
@@ -67,6 +78,19 @@ List<Token> tokenize(String input) {
...
+    if (ch == '`' || ch == '~') {
+      final token = scanner.scanMarkdown();
+      if (token != null) {
+        flushText();
+        tokens.add(token);
+        continue;
+      }
+      // Not a Markdown code region, consume the character as text
+      textBuffer.write(ch);
+      scanner.advance();
```
[the remark this answers is not in the document]
Comment (eunice-hong) [reply lib/src/tokenizer.dart]: Resolved by 757fecda6d8e89730b19c85d8d90c5973844328e

File: lib/src/tokenizer.dart [hunk 2]
```diff
@@ -181,12 +205,102 @@ class _Scanner {
...
+
+    // Search for a closing fence: a line that begins with >= openCount
+    // of the same fence character
+    while (!isDone) {
+      final lineStart = _index;
+      while (!isDone && peek() == fenceChar) {
+        advance();
+      }
+      final closeCount = _index - lineStart;
+      if (closeCount >= openCount) {
```
[the remark this answers is not in the document]
Comment (eunice-hong) [reply lib/src/tokenizer.dart]: Resolved by b86f2edac2eed0c8bfd436a858a3d7e8f72ed636

**Comment by eunice-hong** [comment]
@codex

File: lib/src/tokenizer.dart [hunk 3]
```diff
@@ -181,12 +210,110 @@ class _Scanner {
...
+      advance();
+    }
+    if (!isDone) advance(); // consume '\n'
+
+    // Search for a closing fence: a line that consists of >= openCount fence
+    // characters followed by optional whitespace only (CommonMark §4.4–4.5).
+    while (!isDone) {
+      final lineStart = _index;
+      while (!isDone && peek() == fenceChar) {
+        advance();
```
[the remark this answers is not in the document]
Comment (eunice-hong) [reply lib/src/tokenizer.dart:362]: Resolved by fe0af0d1e81292a1a077a8329808165b2c699189

File: lib/src/tokenizer.dart [hunk 4]
```diff
@@ -181,12 +210,110 @@ class _Scanner {
...
+      }
+      final closeCount = _index - lineStart;
+      // A valid closer must have only optional whitespace after the fence chars.
+      // A line like "```python" has enough backticks but is not a valid closer.
+      bool isValidCloser = closeCount >= openCount;
+      if (isValidCloser) {
+        while (!isDone && peek() != '\n') {
+          if (peek() != ' ' && peek() != '\t') {
+            isValidCloser = false;
```
[the remark this answers is not in the document]
Comment (eunice-hong) [reply lib/src/tokenizer.dart]: Resolved by fe0af0d1e81292a1a077a8329808165b2c699189

**Comment by eunice-hong** [comment]
@codex

File: lib/src/tokenizer.dart [hunk 5]
```diff
@@ -181,12 +210,117 @@ class _Scanner {
...
+      if (openCount < 3) {
+        _index = start;
+        return null;
+      }
+      return _scanFencedBlock(start, fenceChar, openCount);
+    }
+
+    // Backticks: 3 or more start a fenced code block
+    if (openCount >= 3) {
+      return _scanFencedBlock(start, fenceChar, openCount);
```
[the remark this answers is not in the document]
Comment (eunice-hong) [reply lib/src/tokenizer.dart]: Resolved by 9ed9f9345a149820d5457d9eb199a9f7b41a6b7e

**Comment by eunice-hong** [comment]
@codex

File: lib/src/tokenizer.dart [hunk 6]
```diff
@@ -181,12 +210,119 @@ class _Scanner {
...
+        _index = start;
+        return null;
+      }
+      return _scanFencedBlock(start, fenceChar, openCount);
+    }
+
+    // Backticks: 3 or more at a line-start position → fenced code block.
+    // Mid-line runs of 3+ backticks fall through to inline-span scanning.
+    if (openCount >= 3 && _isAtLineStart(start)) {
+      return _scanFencedBlock(start, fenceChar, openCount);
```
[the remark this answers is not in the document]
Comment (eunice-hong) [reply lib/src/tokenizer.dart]: Resolved by 5ee35cd10551100c42d915c95e9137e1f42a41a1

File: lib/src/tokenizer.dart [hunk 7]
```diff
@@ -67,6 +78,24 @@ List<Token> tokenize(String input) {
       continue;
     }
 
+    if (ch == '`' || ch == '~') {
+      final token = scanner.scanMarkdown();
```
[the remark this answers is not in the document]
Comment (eunice-hong) [reply lib/src/tokenizer.dart:82]: Resolved by 5ee35cd10551100c42d915c95e9137e1f42a41a1

**Comment by eunice-hong** [comment]
@codex

File: lib/src/tokenizer.dart [hunk 8]
```diff
@@ -181,12 +210,140 @@ class _Scanner {
...
+      final closeCount = _index - closeStart;
+      if (closeCount == openCount) {
+        return Token(TokenType.markdown, _input.substring(start, _index));
+      }
+      // Count mismatch — keep scanning
+    }
+
+    // No closing delimiter found — backtrack and treat as plain text
+    _index = start;
+    return null;
```
[the remark this answers is not in the document]
Comment (eunice-hong) [reply lib/src/tokenizer.dart]: Resolved by b124dabdf1ec1e1698b4cb3a63d5ef4aecc21df8

**Comment by eunice-hong** [comment]
@codex

**Comment by eunice-hong** [comment]
@codex

File: lib/src/tokenizer.dart [hunk 9]
```diff
@@ -67,6 +78,24 @@ List<Token> tokenize(String input) {
...
+        tokens.add(token);
+        continue;
+      }
+      // scanMarkdown() backtracked — the run of fence characters does not
+      // start a valid code region.  Consume the entire run atomically so
+      // that no sub-run can be re-examined as a new opener; otherwise a
+      // later, shorter sub-run might find a spurious closing delimiter.
+      while (scanner.peek() == ch) {
+        textBuffer.write(ch);
+        scanner.advance();
```
[the remark this answers is not in the document]
Comment (eunice-hong) [reply lib/src/tokenizer.dart]: Resolved by 882c9def4c9d98ce59408ab825639cac6632535c

**Comment by eunice-hong** [comment]
@codex
